pallets/gear/src/benchmarking/syscalls.rs

```diff
@@ -154,10 +154,11 @@ where
         )
     }
 
-    pub fn alloc(r: u32) -> Result<Exec<T>, &'static str> {
+    pub fn alloc(repetitions: u32, pages: u32) -> Result<Exec<T>, &'static str> {
+        assert!(repetitions * pages <= max_pages::<T>() as u32);
```

Comment:
Now we the number if repetitions is `repetitions * API_BENCHMARK_BATCH_SIZE` so please add
```rust
pub fn alloc(batches_amount: u32, pages: u32) -> Result<Exec<T>, &'static str> {
    let repetitions = batches_amount * API_BENCHMARK_BATCH_SIZE;
    assert!(repetitions * pages <= max_pages::<T>() as u32);
```